improver/cli/__init__.py

```diff
@@ -256,7 +256,7 @@ def with_output(wrapped, *args, output=None, **kwargs):
     """
     from improver.utilities.save import save_netcdf
     result = wrapped(*args, **kwargs)
-    if output:
+    if result and output:
```

Comment:
This seems reasonable and probably also a more robust check, Can't see any negative consequences.